SEImplementation/python/sourcextractor/config/measurement_images.py

```diff
@@ -562,33 +562,50 @@ def load_multi_hdu_fits(image_file, psf=None, weight=None, **kwargs):
         single HDU FITS files, one for each weight map
     :return: A ImageGroup representing the images
     """
-    hdu_list = [i for i, hdu in enumerate(fits.open(image_file)) if hdu.is_image and hdu.header['NAXIS'] == 2]
     
-    # handles the PSFs
-    if isinstance(psf, list):
-        assert(len(psf) == len(hdu_list))
-        psf_list = psf
-        psf_hdu_list = [0] * len(psf_list)
-    else:
-        psf_list = [psf] * len(hdu_list)
-        psf_hdu_list = hdu_list
+    if isinstance(image_file, list):
+        assert(isinstance(psf, list) and len(psf) == len(image_file))
+        assert(isinstance(weight, list) and len(weight) == len(image_file))
```

Comment:
I think we should eventually we should raise an exception with a proper message on this cases. Otherwise the users will just gen an assertion error without an explanation of what happened.